app/src/test/java/org/simple/clinic/editpatient/EditPatientInitTest.kt

```diff
@@ -0,0 +1,52 @@
+package org.simple.clinic.editpatient
+
+import com.spotify.mobius.test.FirstMatchers.hasEffects
+import com.spotify.mobius.test.FirstMatchers.hasModel
+import com.spotify.mobius.test.InitSpec
+import com.spotify.mobius.test.InitSpec.assertThatFirst
+import org.junit.Test
+import org.simple.clinic.TestData
+import java.time.format.DateTimeFormatter
+import java.util.Locale
+import java.util.UUID
+
+class EditPatientInitTest {
+  private val patientUuid = UUID.fromString("e40f42f4-0867-4891-ac77-95df5fe1fdef")
+  private val patient = TestData.patient(patientUuid, fullName = "TestName")
+  private val patientAddress = TestData.patientAddress(UUID.fromString("0d19e592-ae4d-4d4f-8881-8764709a17dd"), streetAddress = "Street Address", colonyOrVillage = "Colony")
+  private val patientPhoneNumber = TestData.patientPhoneNumber(UUID.fromString("f204b770-83c5-4145-9ca7-c2273be2bbdc"), number = "9999999999")
+  private val bangladeshNationalId = TestData.businessId(UUID.fromString("c9be5e5d-770c-4f37-a9de-2c304dfebfcd"))
+  private val dateOfBirthFormat = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.ENGLISH)
+  private val initSpec = InitSpec(EditPatientInit(patient, patientAddress, patientPhoneNumber, bangladeshNationalId))
+
+  @Test
+  fun `when edit patient screen is created and colony or villages list is empty, then fetch colony or village list`() {
+    val patientUuid = UUID.fromString("e40f42f4-0867-4891-ac77-95df5fe1fdef")
+    val defaultModel = EditPatientModel.from(patient, patientAddress, patientPhoneNumber, dateOfBirthFormat, bangladeshNationalId, EditPatientState.NOT_SAVING_PATIENT)
+
+    initSpec.whenInit(defaultModel).then(assertThatFirst(
+        hasModel(defaultModel),
+        hasEffects(PrefillFormEffect(patient, patientAddress, patientPhoneNumber, bangladeshNationalId),
+            FetchBpPassportsEffect(patientUuid),
+            LoadInputFields,
+            FetchColonyOrVillagesEffect
+        )
+    ))
+  }
+
+  @Test
+  fun `when edit patient screen is created and has colony or villages list, then don't fetch colony or village list`() {
```

Comment:
We should not be including the details like if the color on the village list empty or not empty in the test. Instead, can we rename it to something like this?

`when screen is restored, then don't fetch colony or villages`